app/src/main/kotlin/com/softteco/template/ui/feature/home/HomeScreen.kt

```diff
@@ -1,59 +1,163 @@
+import androidx.compose.material3.Card
+import androidx.compose.material3.CardDefaults
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
 import androidx.compose.material3.Text
 import androidx.compose.runtime.Composable
-import androidx.compose.runtime.collectAsState
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
 import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextOverflow
 import androidx.compose.ui.tooling.preview.Preview
-import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.compose.ui.unit.dp
 import com.softteco.template.R
 import com.softteco.template.ui.theme.AppTheme
 import com.softteco.template.ui.theme.Dimens
+import com.softteco.template.ui.theme.RoundedCornerSizes
 
 @Composable
 fun HomeScreen(
-    viewModel: HomeViewModel = hiltViewModel(),
-    onApiSampleClicked: () -> Unit = {},
-    onGoLoginClicked: () -> Unit = {},
+    onLoginClicked: () -> Unit = {},
+    onSignatureClicked: () -> Unit = {},
+    onBleClicked: () -> Unit = {},
 ) {
-    ScreenContent(
-        state = viewModel.state.collectAsState().value,
-        onApiSampleClicked = { onApiSampleClicked() },
-        onGoLoggingClicked = { onGoLoginClicked() }
+    val cards = listOf(
+        HomeCards.LoginCard(onLoginClicked),
+        HomeCards.SignatureCard(onSignatureClicked),
+        HomeCards.BleCard(onBleClicked),
     )
+    ScreenContent(cards)
 }
 
 @Composable
 private fun ScreenContent(
-    state: HomeViewModel.State,
+    cards: List<HomeCards>,
     modifier: Modifier = Modifier,
-    onApiSampleClicked: () -> Unit = {},
-    onGoLoggingClicked: () -> Unit = {}
 ) {
-    Box(modifier.fillMaxSize(), contentAlignment = Alignment.Center) {
-        Column(verticalArrangement = Arrangement.spacedBy(Dimens.PaddingSmall)) {
-            Text(state.data)
-            Button(onClick = onApiSampleClicked) {
-                Text("To Api Sample")
-            }
-            Button(onClick = onGoLoggingClicked) {
-                Text(stringResource(id = R.string.go_to_login))
+    LazyVerticalGrid(
+        modifier = modifier,
+        columns = GridCells.Fixed(2),
+        contentPadding = PaddingValues(Dimens.PaddingSmall)
+    ) {
+        items(cards) { item ->
+            HomeCard(
+                title = stringResource(id = item.titleId),
+                description = stringResource(id = item.descriptionId),
+                drawableRes = item.drawableRes,
+                onClick = item.onClick,
+                isEnabled = item.enabled,
+            )
+        }
+    }
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+private fun HomeCard(
+    title: String,
+    description: String,
+    @DrawableRes drawableRes: Int,
+    modifier: Modifier = Modifier,
+    onClick: () -> Unit = {},
```

Comment:
It might be better to make the callback mandatory (to avoid possible miss assignment). 
And make enable=false by default (as a function flag that is enabled only after the implementation is complete)